src/components/articles/createArticle.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from "react";
+import NewArticle from "../../views/articles/createArticle";
+import { connect } from "react-redux";
+import {
+  createNewArticleAction,
+  publishNewArticleAction
+} from "../../actions/articleActions/articleAction";
+import PropTypes from "prop-types";
+import { myHeaders } from "../../utils/myHeaders";
+import { toast } from "react-toastify";
+
+class CreateArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      allcategory: [],
+      title: "",
+      description: "",
+      body: "",
+      category: "",
+      tags: []
+    };
+    this.handleEditorChange = this.handleEditorChange.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    this.handleSelectChange = this.handleSelectChange.bind(this);
+    this.handlePublish = this.handlePublish.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.categories();
+  }
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  };
+
+  handleSelectChange = e => {
+    this.setState({ category: e.target.value });
+  };
+
+  handlePublish = e => {
+    e.preventDefault();
+    this.props.dispatch(
+      publishNewArticleAction(this.props.articles.article.slug)
+    );
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const data = {
+      article: {
+        title: this.state.title,
+        description: this.state.description,
+        body: this.state.body,
+        tags: this.state.tags,
+        category: this.state.category
+      }
+    };
+    this.props.dispatch(createNewArticleAction(data));
+  };
+
+  handleEditorChange = value => {
+    this.setState({ body: value });
+  };
+
+  categories = () => {
+    let initial = [];
+    fetch(
+      "https://ah-backend-lannister-staging.herokuapp.com/api/categories/",
```

Comment:
Use BASE_URL here too

Reply:
I have modified the fetch to use BASE_URL instead